src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/SdkDiffExclusions.txt

```diff
@@ -0,0 +1,17 @@
+# This list is processed using FileSystemName.MatchesSimpleExpression
+#
+# Format
+# {msft|sb},<path> [# comment]
+# msft = Microsoft built SDK
+# sb   = source-built SDK
+#
+# Examples
+# 'folder/*' matches 'folder/' and 'folder/abc'
+# 'folder/?*' matches 'folder/abc' but not 'folder/'
+#
+# We do not want to filter-out folder entries, therefore, we should use: '?*' and not just '*'
+
+msft,./sdk/x.y.z/TestHostNetFramework/?*   # Intentional - MSFT build includes test-host that targets netcoreapp3.1
+msft,./sdk/x.y.z/Sdks/Microsoft.NET.Sdk.WindowsDesktop/?*   # Intentional - explicitly excluded from source-build
+sb,./sdk/x.y.z/TestHost/?*   # Intentional - source-build includes test-host that targets latest .NET TFM
+
```

Comment:
Is there a way this can be utilized to exclude the missing NuGet resources.dll files and include a comment that they are tracked with https://github.com/NuGet/Home/issues/12440?

Reply:
Same applies to the VsTest resources - https://github.com/microsoft/vstest/issues/4305

Reply:
Yes, both of those can be handled. I did not include too many exclusions in the initial commit, to get some discussion first.

Reply:
@MichaelSimons we could use directory-level exclusion to exclude some NuGet and VsTest resources, i.e. the following:

For NuGet pack (one resource file in each of these directories, 13 languages, each for `CoreCLR` and `Desktop`:
```
./sdk/x.y.z/Sdks/NuGet.Build.Tasks.Pack/CoreCLR/cs/?*
./sdk/x.y.z/Sdks/NuGet.Build.Tasks.Pack/Desktop/cs/?*
```
Similarly, the following can be excluded for VsTest (sample language `cs` - there would be 12 more for other languages:
```
./sdk/x.y.z/Extensions/cs/?*
```

However, if we want to cover all files tracked with https://github.com/NuGet/Home/issues/12440, we'd need to add individual file exclusions, like the following (sample language `es` - 12 more would be added):
```
./sdk/x.y.z/es/NuGet.Build.Tasks.Console.resources.dll
./sdk/x.y.z/es/NuGet.Build.Tasks.resources.dll
./sdk/x.y.z/es/NuGet.CommandLine.XPlat.resources.dll
./sdk/x.y.z/es/NuGet.Commands.resources.dll
./sdk/x.y.z/es/NuGet.Common.resources.dll
./sdk/x.y.z/es/NuGet.Configuration.resources.dll
./sdk/x.y.z/es/NuGet.Credentials.resources.dll
./sdk/x.y.z/es/NuGet.DependencyResolver.Core.resources.dll
./sdk/x.y.z/es/NuGet.Frameworks.resources.dll
./sdk/x.y.z/es/NuGet.LibraryModel.resources.dll
./sdk/x.y.z/es/NuGet.Localization.resources.dll
./sdk/x.y.z/es/NuGet.PackageManagement.resources.dll
./sdk/x.y.z/es/NuGet.Packaging.Core.resources.dll
./sdk/x.y.z/es/NuGet.Packaging.resources.dll
./sdk/x.y.z/es/NuGet.ProjectModel.resources.dll
./sdk/x.y.z/es/NuGet.Protocol.resources.dll
./sdk/x.y.z/es/NuGet.Resolver.resources.dll
./sdk/x.y.z/es/NuGet.Versioning.resources.dll
./sdk/x.y.z/es/NuGet.VisualStudio.Contracts.resources.dll
```

I wonder if these would add enough value. They would certainly make the SDK diff smaller, at the expense of much larger exclusions file.

I do understand that the real value in capturing most of the files from SDK diff, in the exclusion file, is a way to to document the difference, by introducing a comment, with valuable reference to a tracking issue or doc.

Similarly, we'd have many individual file instances for additional `VsTest` resources (sample language `de`, with 12 more to be added):
```
./sdk/x.y.z/de/Microsoft.TestPlatform.Build.resources.dll
./sdk/x.y.z/de/Microsoft.TestPlatform.CommunicationUtilities.resources.dll
./sdk/x.y.z/de/Microsoft.TestPlatform.CoreUtilities.resources.dll
./sdk/x.y.z/de/Microsoft.TestPlatform.CrossPlatEngine.resources.dll
./sdk/x.y.z/de/Microsoft.TestPlatform.Utilities.resources.dll
./sdk/x.y.z/de/Microsoft.TestPlatform.VsTestConsole.TranslationLayer.resources.dll
./sdk/x.y.z/de/Microsoft.VisualStudio.TestPlatform.Client.resources.dll
./sdk/x.y.z/de/Microsoft.VisualStudio.TestPlatform.Common.resources.dll
./sdk/x.y.z/de/Microsoft.VisualStudio.TestPlatform.ObjectModel.resources.dll
```

We could alternatively, exclude the whole language directories, i.e the following:
```
 ./sdk/x.y.z/de/?*
```

Most of the resources are coming from NuGet and VsTest, but there are some outliers that we'd be bundling in these larger groups.

Reply:
Is possible and would it make sense to add exclusions that are wildcarded for the language directories and file known patterns?  This is walking a line of reducing the maintenance and capturing why the diff exists yet not letting unintentional changes slip through.

```
# vstest localization is disabled in Linux builds - https://github.com/microsoft/vstest/issues/4305
msft,./sdk/x.y.z/*?/Microsoft.TestPlatform.*?.resources.dll
msft,./sdk/x.y.z/*?/Microsoft.VisualStudio.TestPlatform.*?.resources.dll
msft,./sdk/x.y.z/*?/Test.Utility.resources.dll
msft,./sdk/x.y.z/*?/vstest.console.resources.dll
```

and

`msft,./sdk/x.y.z/*?/NuGet.*?.resources.dll`

Reply:
Yeah, definitely. This helps document the known issues, especially the issues, for which we have filed tracking issues.

Reply:
Fixes with https://github.com/dotnet/installer/pull/15650/commits/d45e89a95f2e84154d9c59a0fbaf91f13f8ff197